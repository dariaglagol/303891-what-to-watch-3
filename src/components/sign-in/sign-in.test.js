import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import SignIn from "./sign-in";
import history from "../../history";

const userErrors = {};

it(`Render sign in`, () => {
  const signInComponent = renderer
    .create(
        <Router history={history}>
          <SignIn
            onSubmit={() => {}}
            userErrors={userErrors}
            invalidFields={[]}
            isLoading={true}
          />
        </Router>
    ).toJSON();

  expect(signInComponent).toMatchSnapshot();
});
