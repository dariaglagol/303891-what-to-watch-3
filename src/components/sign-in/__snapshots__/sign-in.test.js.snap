// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render sign in 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <fieldset
        disabled={true}
      >
        <div
          className="sign-in__fields"
        >
          <div
            className="sign-in__field"
          >
            <input
              className="sign-in__input"
              id="user-email"
              name="user-email"
              pattern="^([a-z0-9_-]+\\\\.)*[a-z0-9_-]+@[a-z0-9_-]+(\\\\.[a-z0-9_-]+)*\\\\.[a-z]{2,6}$"
              placeholder="Email address"
              required={true}
              type="email"
            />
            <label
              className="sign-in__label visually-hidden"
              htmlFor="user-email"
            >
              Email address
            </label>
          </div>
          <div
            className="sign-in__field"
          >
            <input
              className="sign-in__input"
              id="user-password"
              name="user-password"
              pattern="[A-Za-z0-9]+$"
              placeholder="Password"
              required={true}
              type="password"
            />
            <label
              className="sign-in__label visually-hidden"
              htmlFor="user-password"
            >
              Password
            </label>
          </div>
        </div>
        <div
          className="sign-in__submit"
        >
          <button
            className="sign-in__btn"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </fieldset>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
